### PR TITLE
Add a script to format all version-controlled source files with astyle. 

### PR DESCRIPTION
**Pull request by haydenroche5**
Also, add a GitHub action to ensure all commits use properly formatted code.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,3 +14,12 @@ jobs:
         uses: actions/checkout@v3
       - name: Run unit tests
         uses: ./.github/actions/run_unit_tests
+  run_astyle:
```
Comment (haydenroche5) [inline .github/workflows/ci.yml:17]: I want to Dockerize this, too, eventually, just as we did for running the unit tests. However, we aren't set up with a container registry, yet, and I want this job to run and give feedback quickly. That's at odds with building the image anew on every run. So, for now, I'm not Dockerizing this one.

File: n_ftoa.c [hunk 2]
```diff
@@ -287,8 +287,7 @@ fmtflt(char *str, size_t *len, size_t size, JNUMBER fvalue, int width,
         if (fpos > 0)	/* Omit trailing fractional part zeros. */
             while (omitcount < fpos && fconvert[omitcount] == '0') {
                 omitcount++;
-            }
-        else {	/* The fractional part is zero, omit it completely. */
+            } else {	/* The fractional part is zero, omit it completely. */
```
Comment (zfields) [inline n_ftoa.c:290]: tabs / spaces

**Review by zfields — APPROVED** [review]

File: n_ftoa.c (hunk 2 above)
Comment (zfields) [reply n_ftoa.c:290]: which ironically seems like it should have been picked up by `astyle`... :raised_eyebrow:

Comment (haydenroche5) [reply n_ftoa.c]: Fixed and added braces to the opening `if`, which I think makes the code clearer.

**Comment by zfields** [comment]
Merge it already.

**Pull request merged**